pkg/config/marshal.go

```diff
@@ -0,0 +1,124 @@
+package config
+
+import (
+	"crypto/x509"
+	"io/ioutil"
+	"strconv"
+
+	toml "github.com/pelletier/go-toml"
+	"github.com/spf13/cast"
+)
+
+const (
+	keyURL            = "core.dcos_url"
+	keyACSToken       = "core.dcos_acs_token"
+	keyTLS            = "core.ssl_verify"
+	keyTimeout        = "core.timeout"
+	keySSHUser        = "core.ssh_user"
+	keySSHProxyIP     = "core.ssh_proxy_ip"
+	keyPagination     = "core.pagination"
+	keyReporting      = "core.reporting"
+	keyMesosMasterURL = "core.mesos_master_url"
+	keyPrompLogin     = "core.prompt_login"
+	keyClusterName    = "cluster.name"
+)
+
+// FromPath creates a Config based on a path to a .toml file.
+func FromPath(path string) (Config, error) {
+	tree, err := toml.LoadFile(path)
+	if err != nil {
+		return Config{}, err
+	}
+	conf := fromStore(NewStore(tree, StoreOpts{}))
+	conf.path = path
+	return conf, nil
+}
+
+// FromString creates a Config using a string representing the configuration formatted as a TOML document.
+func FromString(tomlData string) (Config, error) {
+	tree, err := toml.Load(tomlData)
+	if err != nil {
+		return Config{}, err
+	}
+	return fromStore(NewStore(tree, StoreOpts{})), nil
+}
+
+func fromStore(store *Store) Config {
+	conf := DefaultConfig()
+	Unmarshal(&conf, store)
+	conf.store = store
+	return conf
+}
+
+func Unmarshal(conf *Config, store *Store) {
+	for _, key := range store.Keys() {
+		val, _ := store.Get(key)
+		switch key {
+		case keyURL:
+			conf.URL = cast.ToString(val)
+		case keyACSToken:
+			conf.ACSToken = cast.ToString(val)
+		case keyTLS:
+			conf.TLS = UnmarshalTLS(val)
+		case keyTimeout:
+			conf.Timeout = cast.ToInt(val)
+		case keySSHUser:
+			conf.SSHUser = cast.ToString(val)
+		case keySSHProxyIP:
+			conf.SSHProxyIP = cast.ToString(val)
+		case keyPagination:
+			conf.Pagination = cast.ToBool(val)
+		case keyReporting:
+			conf.Reporting = cast.ToBool(val)
+		case keyMesosMasterURL:
+			conf.MesosMasterURL = cast.ToString(val)
+		case keyPrompLogin:
+			conf.PrompLogin = cast.ToBool(val)
+		case keyClusterName:
+			conf.ClusterName = cast.ToString(val)
+		}
+	}
+}
+
+// UnmarshalTLS creates a TLS struct from a string or bool-like string.
+//
+// Valid values are:
+// - A path to a root CA bundle.
+// - "1", "t", "T", "TRUE", "true", "True" - will use the system's CA bundle.
+// - "0", "f", "F", "FALSE", "false", "False" - will send insecure requests.
+func UnmarshalTLS(val interface{}) TLS {
+	strVal := cast.ToString(val)
+	if strVal == "" {
+		return TLS{}
+	}
+
+	// Try to cast the value to a bool, true means we verify
+	// server certificates, false means we skip verification.
+	if verify, err := strconv.ParseBool(strVal); err == nil {
+		return TLS{
+			Insecure: !verify,
+		}
+	}
+
+	// If the value is not a bool-like string, it means it's a path to a root CA bundle.
+	rootCAsPEM, err := ioutil.ReadFile(strVal)
+	if err != nil {
+		return TLS{
+			Insecure:    true,
+			RootCAsPath: strVal,
+		}
+	}
+
+	certPool := x509.NewCertPool()
```

Comment:
What are we doing here? It is explained in `config.go` that it is a `pool of root CAs to verify server certificates against.` but writing for which reasons would we arrive in that part of the function and what does it imply could be useful.